packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,221 @@
+`;
+
+export interface DeviceDisplayProps {
+    address: string;
+    network?: string;
+}
+
+export const DeviceDisplay = ({ address, network }: DeviceDisplayProps) => {
+    const selectedDeviceInternalModel = useSelector(selectDeviceInternalModel);
+    const unavailableCapabilities = useSelector(selectDeviceUnavailableCapabilities);
+    const addressDisplayType = useSelector(selectAddressDisplayType);
+
+    const isCardano = network === 'cardano';
+    const areChunksUsed =
+        addressDisplayType === AddressDisplayOptions.CHUNKED &&
+        !unavailableCapabilities?.chunkify &&
+        !isCardano;
+    const isPixelType = selectedDeviceInternalModel !== DeviceModelInternal.T2T1;
+    const iconSize = isPixelType ? 10 : 20;
+    const iconColor = isPixelType ? '#ffffff' : '#959596';
+    const iconNextName = isPixelType ? 'ADDRESS_PIXEL_NEXT' : 'ADDRESS_NEXT';
+    const iconContinuesName = isPixelType ? 'ADDRESS_PIXEL_CONTINUES' : 'ADDRESS_CONTINUES';
+
+    const showChunksInRows = (chunks: ReactNode[][] | undefined, isNextAddress?: boolean) =>
+        chunks?.map((row, rowIndex) => (
+            <Row
+                // eslint-disable-next-line react/no-array-index-key
+                key={rowIndex}
+                isAlignedRight={rowIndex === 0 && isNextAddress}
+            >
+                {row.map((chunk, chunkIndex) => (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const renderChunks = (address: string) => {
+        const chunks = address.match(/.{1,4}/g);
+
+        if (isCardano) {
+            // Add pagination icons
+            // @ts-expect-error
+            chunks?.splice(15, 0, <Icon size={iconSize} icon={iconNextName} color={iconColor} />);
+            chunks?.splice(
+                16,
+                0,
+                // @ts-expect-error
+                <Icon size={iconSize} icon={iconContinuesName} color={iconColor} />,
+            );
+        }
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            result[rowIndex].push(chunk);
+
+            return result;
+        }, [] as ReactNode[][]);
+
+        const handleOnCopy = (event: ClipboardEvent) => {
+            const selectedText = window.getSelection()?.toString();
+
+            if (selectedText) {
+                const processedText = selectedText.replace(/\s/g, '');
+                event?.nativeEvent?.clipboardData?.setData('text/plain', processedText);
+                event.preventDefault();
+            }
+        };
+
+        // this is now not possible but will be used once Cardano flow is improved
+        if (isCardano) {
```

Comment:
I would prefer not to add dead code because it tends to stay forever. But I would not consider it a blocker for this PR. Ideally, I'd take it out of this PR during rebase and prepare separate branch/draft PR for cardano.

Reply:
Will removed it from this PR and create a new one from this where Cardano is expected to work on FW so we also can show it in Suite